kotlin/internal/jvm/compile.bzl

```diff
@@ -389,37 +514,34 @@ def _run_kt_builder_action(ctx, rule_kind, toolchains, dirs, srcs, friend, compi
     )
 
     ctx.actions.run(
-        mnemonic = "KotlinCompile",
+        mnemonic = mnemonic,
         inputs = depset(
-            ctx.files.srcs,
+            ctx.files.srcs + generated_src_jars,
             transitive = [compile_deps.compile_jars, transitive_runtime_jars],
         ),
         tools = tools,
         input_manifests = input_manifests,
         outputs = [f for f in outputs.values()],
         executable = toolchains.kt.kotlinbuilder.files_to_run.executable,
-        execution_requirements = {"supports-workers": "1"},
+        execution_requirements = {
+            "supports-workers": "1",
+            "worker-key-mnemonic": "KotlinCompile"
```

Comment:
This means `KotlinCompile` and `KotlinKAPT` share the same workers. I'd like to support Multiplexed workers too. It looks like JavaBuilder supports multiplexed workers in Bazel 3.7 and someone submitted maybe a fix for the "multiplexed workers sometimes hang" bug reported by some users which I guess will make it into Bazel 3.8

Reply:
I'm working on multiplexer support. It has some pitfalls. The way kt workers are intialized is... interesting.